File: ios/TechBlogCatchup/Models/Job.swift
import Foundation

struct JobInfo: Codable, Identifiable, Hashable {
    let id: Int
    let jobType: String
    let status: String
    let params: String?
    let result: String?
    let errorMessage: String?
    let createdAt: Date
    let startedAt: Date?
    let completedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id, status, params, result
        case jobType = "job_type"
        case errorMessage = "error_message"
        case createdAt = "created_at"
        case startedAt = "started_at"
        case completedAt = "completed_at"
    }
}

File: android/lib/models/job.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'job.freezed.dart';
part 'job.g.dart';

@freezed
class Job with _$Job {
  const factory Job({
    required int id,
    required String jobType,
    required String status,
    String? params,
    String? result,
    String? errorMessage,
    required DateTime createdAt,
    DateTime? startedAt,
    DateTime? completedAt,
  }) = _Job;

  factory Job.fromJson(Map<String, dynamic> json) => _$JobFromJson(json);
}
